### PR TITLE
stylo bindings update

### PR DESCRIPTION
**Pull request by heycam**
For the latest m-c stylo sync, on top of #14472.

r? @emilio

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14492)

### CODE REVIEW

File: components/style/binding_tools/regen.py [hunk 1]
```diff
@@ -106,6 +106,7 @@
             "AnonymousContent",
             "AudioContext",
             "CapturingContentInfo",
+            "ConsumeStyleBehavior",
```
Comment (emilio) [inline components/style/binding_tools/regen.py]: These should be noops, right? They're in the mozilla namespace above. Anyway, happy to clean it up afterwards.

**Review by emilio — APPROVED** [review]

File: components/style/gecko_bindings/structs_debug.rs [hunk 2]
```diff
@@ -1176,7 +1175,10 @@ pub mod root {
         pub struct ThreadSafeAutoRefCnt {
             pub mValue: u64,
         }
-        pub const ThreadSafeAutoRefCnt_isThreadSafe: bool = true;
+        extern "C" {
+            #[link_name = "_ZN7mozilla20ThreadSafeAutoRefCnt12isThreadSafeE"]
+            pub static ThreadSafeAutoRefCnt_isThreadSafe: bool;
+        }
```
Comment (emilio) [inline components/style/gecko_bindings/structs_debug.rs]: Yeah, you're definitely using clang 3.8, but if you're not building with `--features llvm_stable`, so I expect stuff to break, like this, and potentially the `JSWhyMagic` thing.

I'm actually surprised it works so decently to be fair, but that's good :)

File: components/style/binding_tools/regen.py (hunk 1 above)
Comment (heycam) [inline components/style/binding_tools/regen.py]: For some reason without adding these namespace-less names here, the types didn't get output.

File: components/style/gecko_bindings/structs_debug.rs (hunk 2 above)
Comment (heycam) [inline components/style/gecko_bindings/structs_debug.rs]: I am using `--features llvm_stable`.  Does it make sense to make regen.sh require to a newer clang?

**Comment by emilio** [comment]
On Thu, Dec 08, 2016 at 11:51:16AM -0800, Cameron McCormack wrote:
> I am using `--features llvm_stable`.  Does it make sense to make regen.sh require to a newer clang?

I guess eventually, yes, but for now it's fine.

**Pull request merged**